client/src/components/Drawer/UseDrawer/UseDrawer.hook.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useMemo, useState } from 'react'
+
+export type AnchorVariant = {
+  top: boolean
+  left: boolean
+  bottom: boolean
+  right: boolean
+}
+
+export type UseDrawerReturnOptions = {
+  toggle: (anchor: keyof AnchorVariant, open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => void
+  side: AnchorVariant
+}
+
+
+/**
+ * Hook descriptions
+ *
+ * @example
+ * const options = useDrawer(conf)
+ */
+export const useDrawer = (): UseDrawerReturnOptions => {
+  const [side, setSide] = useState<AnchorVariant>({
```

Comment:
I whould prefer have one prop `side?: 'left' | 'right' | 'top' | 'bottom'` and default value as `right`